[Docs/Reporting] Improve docs on automating Reporting (#40931)
elastic/kibana: docs/reporting/script-example.asciidoc

To automatically generate reports from a script, you'll make a request to the POST URL.

> The response from this request with be JSON, and will contain a `path` property with a

Reviewer: ```suggestion
The response from this request will be JSON, and will contain a `path` property with a
```

<4> The POST URL. You can copy and paste the URL for any report from the Kibana UI.

Here is an example response:

[source,json]
---------------------------------------------------------
{
  "path": "/api/reporting/jobs/download/jxzaofkc0ykpf4062305t068", <1>
  "job": {
    "id": "jxzaofkc0ykpf4062305t068",
    "index": ".reporting-2018.11.11",
    "jobtype": "csv",
    "created_by": "elastic",
    "payload": ..., <2>
    "timeout": 120000,
    "max_attempts": 3,
    "priority": 10
  }
}
---------------------------------------------------------
// CONSOLE

<1> The relative path on the Kibana host for downloading the report.
<2> (Not included in the example) Internal representation of the reporting job, as
found in the `.reporting-*` index.

[IMPORTANT]
===================
When using the `path` to request the download, you will get a 503 (Service Unavailable)
response if report generation hasn't completed yet. In that case, retry after the
number of seconds in the `Retry-After` header in the download API response until the
report is complete.

If there was an error in generating the report, the download URL will return a 500

> (Internal Server Error) response. More information will be available in the the

Reviewer: ```suggestion
(Internal Server Error) response. More information is available in the
```

> Reporting management page in Kibana: *Management > Kibana > Reporting*

Author: Info about 503 was pre-existing. Info about the 500 is new
